### DISCUSSION TITLE
Gestion HC-HP

### DISCUSSION THREAD

**Discussion by Tony-2106** [FredM67/PVRouter-3-phase#10] [General]
Bonjour Fred,

Tout d'abord un grand merci pour le travail accompli et le partage des ces informations.
Je souhaite avoir votre retour sur l'utilisation de votre installation et la gestion des HC-HP.

Je suis aujourd'hui équipé de PV Enphase en autoconsommation avec Rejet de surplus gratuitement sur le réseau. C'est pour cela que je souhaite installer un routeur solaire pour diriger mon surplus vers mon BEC. Celui est un Sauter Prodigio 300l, ballon d'eau chaude dis "intelligent", c'est à dire que celui chauffe suivant notre consommation quotidienne et le nombre de personnes dans la maison, il s'adapte tout seul.
Aujourd'hui, il chauffe sur les périodes heures creuses (22h-6h).
Ma question, comment va se comporter mon BEC, car celui aujourd'hui a "compris" quand sont mes HC et chauffe en conséquences de l'eau qui a été consommé en journée ? 
Si en journée mon BEC est chauffé grâce au routeur, arrivé en HC, mon BEC complémentera la chauffe si celle-ci n'a pas été assez longue ? 
Qu'en est il du mode contre la légionelle, (Cette fonction empêche le développement de bactéries en chauffant l'ensemble du ballon à 65°C. Elle se déclenche si la température de stockage de l’eau ne dépasse pas 60°C pendant trois semaines ) ?
 
Dernière question, Sur différents sites, j'ai pu voir que les routeurs solaires n'étaient pas conseillés avec des résistances stéatite, est ce vrai ?

Si je ne dis pas de bêtises, vous avez une installation similaire à la mienne....

Merci d'avance,

Tony

**Comment by FredM67** [comment]
Bonjour Tony,

> Tout d'abord un grand merci pour le travail accompli et le partage des ces informations. Je souhaite avoir votre retour sur l'utilisation de votre installation et la gestion des HC-HP.

De rien ;-) Au moins, cela fait plaisir que cela puisse servir à d'autres que moi ...

> Je suis aujourd'hui équipé de PV Enphase en autoconsommation avec Rejet de surplus gratuitement sur le réseau. C'est pour cela que je souhaite installer un routeur solaire pour diriger mon surplus vers mon BEC. Celui est un Sauter Prodigio 300l, ballon d'eau chaude dis "intelligent", c'est à dire que celui chauffe suivant notre consommation quotidienne et le nombre de personnes dans la maison, il s'adapte tout seul. Aujourd'hui, il chauffe sur les périodes heures creuses (22h-6h). Ma question, comment va se comporter mon BEC, car celui aujourd'hui a "compris" quand sont mes HC et chauffe en conséquences de l'eau qui a été consommé en journée ?

Je doute très fortement que le BEC ait trouvé "tout seul" la période d'HC. Il est très sûrement relié au contacteur du compteur ou recoit un signal de quelque part.
J'ai survolé la doc, on peut dire que c'est un bon truc marketing, mais techniquement parlant, ce n'est rien d'autre qu'un chauffe-eau ACI !

```
• un système innovant qui mesure le volume d'eau chaude restant ;
• un thermostat électronique de nouvelle génération qui communique avec une commande mobile.
```
Lol, on dirait que ce chauffe-eau a été concu par la NASA !

> Si en journée mon BEC est chauffé grâce au routeur, arrivé en HC, mon BEC complémentera la chauffe si celle-ci n'a pas été assez longue ? Qu'en est il du mode contre la légionelle, (Cette fonction empêche le développement de bactéries en chauffant l'ensemble du ballon à 65°C. Elle se déclenche si la température de stockage de l’eau ne dépasse pas 60°C pendant trois semaines ) ?

Si le cablage entre routeur, contacteur HC et chauffe-eau est fait correctement (cela nécessitera de tirer plus fils supplémentaires), et éventuellement que la partie "programmation" du module de commande du chauffe-eau est adéquate, tout devrait fonctionner de facon optimale.
Pour cela, il faudra que je lise la doc précisément :D

> 
> Dernière question, Sur différents sites, j'ai pu voir que les routeurs solaires n'étaient pas conseillés avec des résistances stéatite, est ce vrai ?

Oui, ce sont des "on-dit"... Il peut arriver que les résistances vibrent, mais je n'ai jamais entendu parlé de vrais problèmes. J'ai un chauffe-eau triphasé de 6.6kW (6 résistances stéatiques de 1.1kW), et tout fonctionne parfaitement depuis longtemps.

> 
> Si je ne dis pas de bêtises, vous avez une installation similaire à la mienne....

Presque... j'ai un chauffe-eau "de base", sans aucune électronique, avec thermostat mécanique. Pour les panneaux, oui, j'ai une installation Enphase avec un peu plus de 6kWc, avec revente de surplus.

> 
> Merci d'avance,
> 
> Tony

Fred

**Reply by Tony-2106** [reply]
Oui on est d'accord, il y a une grosse partie de marketing derrière tout ça.
Effectivement, pour les HC, je le gère via un module Shelly piloté par Jeedom (ce qui reproduit un contacteur HC-HP de tableau).
Quand je dis "intelligent", je veux dire qu'il y a un quand même une gestion de chauffe. Car suivant si nous recevons du monde ou pas je me rends compte que le temps de chauffe est plus ou moins long, et que la consommation est bcp plus importantes quand bcp de monde sont à la maison. Mais c'est sûrement ce que doit faire un BEC classique ?! non

Donc si je résume bien, en journée, lorsque je suis en surplus de production, mon chauffe eau recevras la puissance correspondant à mon surplus et chauffera direct, alors qu'en soirée il fonctionnera de manière classique et chaufferas si celui-ci se rend compte qu'il n'est pas assez chaud, on est d'accord ?!

Merci,

Tony

**Comment by FredM67** [comment]
> Oui on est d'accord, il y a une grosse partie de marketing derrière tout ça. Effectivement, pour les HC, je le gère via un module Shelly piloté par Jeedom (ce qui reproduit un contacteur HC-HP de tableau).
> Quand je dis "intelligent", je veux dire qu'il y a un quand même une gestion de chauffe. Car suivant si nous recevons du monde ou pas je me rends compte que le temps de chauffe est plus ou moins long, et que la consommation est bcp plus importantes quand bcp de monde sont à la maison. Mais c'est sûrement ce que doit faire un BEC classique ?! non

Non, un BEC classique chauffe dès que le thermostat déclenche. Je pense que ce système enregistre simplement les températures au cours du temps, et en déduit telle ou telle situation.
Je n'ai jamais trouvé le schéma complet d'une carte ACI. Je pense fortement que le connecteur sur une extrémité permet via une commande externe, de couper la chauffe "en avance" avant que le thermostat de la carte qui se situe sous la carte de puissance ne coupe.
On peut donc réguler au fil du temps. Le thermostat "physique" est très certainement régler au max.

> 
> Donc si je résume bien, en journée, lorsque je suis en surplus de production, mon chauffe eau recevras la puissance correspondant à mon surplus et chauffera direct, alors qu'en soirée il fonctionnera de manière classique et chaufferas si celui-ci se rend compte qu'il n'est pas assez chaud, on est d'accord ?!

Je pense qu'il faudra faire une sorte de by-pass de la commande de régulation, sinon le chauffe-eau ne chauffera pas !

> 
> Merci,
> 
> Tony

**Reply by FredM67** [reply]
En regardant de plus près les photos dans le doc, la régulation "habituelle" est en fait remplacée par la régulation "intelligente". La carte de puissance, elle, reste la même.
Dans ces conditions, je ne vois pas trop d'autres solutions que régler la programmation sur "Confort", le plus chaud possible, histoire de ne pas "brider" le routeur. Ensuite, pour éviter que ca chauffe aussi au max la nuit et/ou de facon inutile avec de l'électricité "non gratuite", il faudra bidouiller un truc avec un Shelly ou quelque chose comme ca.

**Reply by Tony-2106** [reply]
Merci pour ton retour. Je vais réfléchir à tout ça !
J'ai la possibilité de by-passé la partie commande en mettant le chauffe eau en mode manuel (et de ne plus utiliser l'automatisation de la télécommande comme indiqué page 10 de la doc).
Comme ça je pourrais gérer le temps de chauffe avec le Shelly et il faudrait juste que j'arrive à récupérer une valeur de température pour automatiser tout ça....
Ou partir sur un autre BEC...

**Reply by FredM67** [reply]
Il existe des Shelly avec possibilité de brancher un capteur de température...

**Reply by Tony-2106** [reply]
Oui, c'est ce que j'utilise pour automatiser la pompe de ma piscine ! 😉
Je vais étudier ça ! 
Merci

**Reply by Tony-2106** [reply]
Bonjour Fred, 

Comme tu vois j'ai mis un peu de temps pour étudier tout ça ! ;-)
Je me permet de relancer le sujet, donc j'ai regarder ton wiki pour la partie câblage et je voulais savoir à quoi sert le contacteur Hager HSC 225 que tu mets en sortie du routeur solaire ?

Merci d'avance

**Reply by FredM67** [reply]
Salut Tony,
Il faut toujours prendre son temps ;-)
Tu parles de quel schéma, car j'en ai plusieurs avec ce contacteur...
Fred

**Reply by Tony-2106** [reply]
Ah oui excuse moi !

https://github.com/FredM67/PVRouter-3-phase/blob/main/docs/CarteACI_Mono_Autre_Routeur.pdf

page 2 ;-)

**Reply by FredM67** [reply]
Ah ok.
Certains routeurs alimentent leur sortie via leur propre alimentation. Étant donné qu'il faille dans tous les cas alimenter la carte ACI par du courant permanent, il ne peut pas l'alimenter via ce genre de routeur.
Et on ne peut pas non plus intercaler ce genre de routeur entre la carte ACI et les résistances, car le routeur ne sera plus alimenté dès que le chauffe-eau est à température.
Il faut donc faire un "2ième" circuit.
- circuit 1 alimente la carte ACI avec du courant permanent.
- circuit 2 provient de la sortie du routeur, alimenté aussi en permanent. En sortie, on a du courant haché.
- le contacteur "réplique" la commande du thermostat de la carte pour éviter la sur-chauffe. Un tel contacteur a 2 circuits séparés, celui qui alimente la bobine, et l'autre circuit de puissance contrôlé par la bobine. Ce contacteur DOIT être ouvert au repos de telle sorte qu'en cas de panne quelconque, il est garanti que les résistances ne seront pas alimentées.
Si on recycle un contacteur HC pour faire le job, il ne faudra JAMAIS le mettre sur position 1.

**Comment by Tony-2106** [comment]
Bonjour Fred,

c'est encore moi !
Je continu de potasser tout ça ! ;-)
Donc dans mon cas, si je suis tes préconisations de branchement, la carte ACI sera tout le temps alimentée, le chauffe eau "ne comprendra plus les HC et HP". 
Aujourd'hui lorsque je suis en HP (donc chauffe eau non alimenté, il y a une petite batterie qui prend le relais pour garder la carte  ACI sous tension). 
Je me pose donc la question, si je continu d'alimenter la carte ACI via mon contacteur HC-HP (module Shelly dans mon cas),  et que j'alimente les résistances par le routeur solaire comme tu l'as expliqué dans ton schéma.
Je me dis comme ça en journée lors de production solaire, je chauffe mon ballon et en nuit lors des HC, la carte ACI prend le relais et chauffe suivant nos habitudes de consommation.
As ton avis est ce que ça fonctionnerait ?
De cette façon je garderai la "partie intelligente" du chauffe eau ? non ?

Et dernière question, comment tout cela ce comporte en journée lorsqu'il y a un très gros excédent de production solaire qui est dirigé vers le chauffe eau ? c'est à dire que si celui-ci est arrivé à température et qu'il n'a plus besoin d'être chauffé, le routeur comprends qu'il n'a plus besoin d'envoyer de la puissance au chauffe eau ?

Merci d'avance,

**Reply by FredM67** [reply]
Hello Tony,
Presque ;-)
Oui, la carte ACI sera alimentée en permanence.
Non, "ne comprendra plus les HC et HP". Le câblage est différent. Le contacteur HC agira sur le circuit de puissance qui a été "débarassé" de la carte ACI.
Il faut **impérativement** que le circuit de puissance passe d'une manière ou d'une autre par les relais de la carte ACI, pour des raisons de sécurité.

Lorsque la temérature est atteinte, la carte ACI ouvert les relais, et les résistances ne sont plus alimentées. C'est pour cela qu'il FAUT que le circuit de puissance passe TOUJOURS par la carte ACI.
Le routeur continuera d'alimenter le circuit de puissance, qui sera coupé par les relais de la carte ACI... Même fonctionnement que chauffe pendant les HC.

**Reply by FredM67** [reply]
https://github.com/FredM67/PVRouter-3-phase/blob/main/docs/HowToInstall.pdf
Schéma "AVEC contacteur Jour/Nuit et
carte ACI monophasée"... selon ta carte ACI, kitable/non-kitable.

**Reply by Tony-2106** [reply]
Merci pour ta réactivité !
Je commence à comprendre tout ça ! ;-)

Question de sécurité, si je suis ton schéma 
https://github.com/FredM67/PVRouter-3-phase/blob/main/docs/CarteACI_Mono_Autre_Routeur.pdf  (page 2)

Que se passe-t-il, si le routeur solaire envoie de la puissance et que contacteur HC-HP est sur est sur marche forcé ?

**Reply by FredM67** [reply]
Rien... le circuit de puissance est contrôlé par la carte ACI via le contacteur "standard"... Si on est à température, la carte ACI coupe le contacteur, et donc les résistances ne peuvent plus être alimentées, peu importe le reste (routeur, contacteur HC)

**Reply by Tony-2106** [reply]
ok top  ! merci pour ton retour !

**Comment by Tony-2106** [comment]
Salut Fred,

Je viens de commander mon routeur, j'ai pris le Arsun.
Sur le site de celui-ci le plan du câblage diffère un peu avec le tien ? 

https://www.arsun.fr/modif-thermostat.html

Même si je ne pense pas qu'il est une grande importance mais les câbles rouges sont connectés au A2 et inversement pour les câbles bleus qui sont connectés au A1.
Là, ou c'est différent, c'est que Arsun et la carte ACI ne sont reliés que part A1 et A2 du contacteur,  et les sorties du contacteurs sont envoyés à la résistance du chauffe eau. Il n'utilise pas de wago pour rassembler le tout. 
Serais tu me l'expliquer ?

Merci d'avance,

**Reply by FredM67** [reply]
Sniff... pourquoi pas un mk2pvrouter ?
Techniquement, le schéma est le même, à condition de tout bien respecter entre phase et neutre...
J'ai mis à jour pour faire 2 circuits indépendants.

**Reply by Tony-2106** [reply]
je t'avouerai que j'ai hésité pendant longtemps... Je suis allé à la facilité !
En tou cas, encore merci pour ta disponibilité.

Je viendrai alimenter la discussion une fois tout mis en place pour faire un retour, si ça peut servir pour d'autres personnes !

**Comment by ApiFelix** [comment]
bonjour Fred et merci pour ton travail

Pour ma part j'ai choisi mk2pvrouter !!
avec ton code

je suis en Tri avec BEC mono thermostat mécanique 

j'ai un pb métaphysique : dans la famille on se douche plutôt le soir. Et avec des HC/CP de 2h à 7h .. mon BEC va chauffer la nuit .. et sera chaud lorsqu'il y aura du soleil ...

Je pensais supprimer le contacteur et passer en "marche forcée" via le soft du routeur mais as tu une meilleur idée ?

merci

**Reply by FredM67** [reply]
Salut,
Ton problème métaphysique est en réalité le pb que 80-90% des gens ont :D
Pour cela il y a 2 solutions, dont une quasi parfaite :D
La 1ère est "décrite" [ici](https://github.com/FredM67/PVRouter-3-phase/tree/main/Mk2_3phase_RFdatalog_temp#configuration-de-la-gestion-des-heures-creuses-dual-tariff).
L'inconvénient majeur de cette solution est l'absence de connaissance du lendemain, à savoir, est-ce qu'il y aura du soleil ou pas !
La solution quasi parfaite est celle qui combine le routeur (et le prog modifié en conséquence, ce qui est déjà prévu) et de la domotique.
Avec HomeAssistant par exemple, il est possible d'obtenir les prévisions d'ensoleillement et même de production solaire de facon +/- fiable (site solcast par exemple). Tu peux alors demander une marche forcée en fonction de cette info ainsi que la température de ton chauffe-eau à un moment donné. Si HA tourne sur un Raspberry à proximité du routeur, on peut relier les 2 par quelques fils pour piloter le routeur, mais on peut aussi installer dans le boitier du routeur un ESP32 ou truc du genre qui recevra les ordres d'HA, et pilotera en conséquence le routeur.
Dans l'absolu, le choix de la meilleure solution dépend de chacun, ce qu'on a déjà à disposition en matière de domotique, où sont installés tel ou tel appareil (routeur, raspberry, controle par Wifi ou par câble, ...) et surtout du degré d'optimisation que l'on souhaite atteindre, enfin voilà, comme toujours, les neurones vont fumer :D

Fred

**Reply by ApiFelix** [reply]
merci de ta réponse
J'ai HA sur Raspberry, mais je fais moyennement confiance aux prévisions météo !
Je pensais sinon à un "compteur de watt" envoyés par le routeur : le routeur incrémente un index des watt envoyés sue la sortie du BEC. à 2h du mat je regarde cet index . Si il est inférieur à une consigne (par ex 1000w) c'est que le BEC n'a pas beaucoup chauffé dans la journée et je passe en MF

sur le papier c'est applicable.  C'est réaliste au niveau du routeur ?
merci

**Reply by FredM67** [reply]
Solcast, ce sont des prévisions de radiation solaire. Ce site est génial, car il donne des prédictions avec marge d'erreur.
Tu as ce genre de données par tranche de 30 mn:
```json
        {
            "pv_estimate": 2.3175,
            "pv_estimate10": 1.3151,
            "pv_estimate90": 2.8387,
            "period_end": "2024-03-21T09:00:00.0000000Z",
            "period": "PT30M"
        },
```
Comme toutes les prévisions, moins tu regardes dans le futur, plus c'est précis. Vers la fin de tes heures creuses, tu peux donc estimer de façon correcte si tu pourras chauffer ou pas ton ECS. Evidemment, ce genre de truc sera à affiner pour avoir un fonctionnement "optimal".
Ton calcul de Watt sera bien plus approximatif, d'autant plus que tu ne pourras pas tenir compte de la quantité d'eau utilisée. Juste en hiver pour moi, si je tire 10l d'eau chaude cela n'a rien à voir avec 10l tiré l'été, car l'eau froide est bcp plus froide en hiver. Le seul paramètre fiable est la température de l'eau.
Sur le mono, il y a déjà un compteur de Watt routé, mais sur le tri, c'est mort, techniquement impossible.
Fred

**Reply by ApiFelix** [reply]
Merci là au moins c'est clair !!!
tu as déjà implémenté l'esp32 dans le routeur ? je partirai bien la dessus
merci

**Reply by FredM67** [reply]
Il n'y a pas d'implémentation spéciale. J'ai déjà une interface de base, mais rien n'est prévu pour HomeAssistant.
Tu peux prendre n'importe quoi comme ESP32, ensuite, il faudra câble une ou plusieurs sorties sur le routeur pour commander telle ou telle fonction.
